src/data/queries/headerFooter.ts

```diff
@@ -7,82 +7,68 @@ import {
 import { queries } from '.'
 import { drupalClient } from '@/lib/drupal/drupalClient'
 import { Menu, HeaderMegaMenu } from '@/types/dataTypes/drupal/menu'
-import { HeaderFooterData, MegaMenuPromoColumn } from '@/types/index'
+import { HeaderFooterData } from '@/types/index'
 import { buildHeaderFooterData } from '@/lib/utils/headerFooter'
 
 export type RawHeaderFooterData = {
   footerColumns: Menu
   footerBottomRail: Menu
   headerMegaMenu: HeaderMegaMenu
-  promoBlocks: MegaMenuPromoColumn[]
 }
 
 // Define the query params for fetching footer menu data.
 export const params: QueryParams<null> = () => {
   return queries.getParams().addFields('menu_items', ['title,url'])
 }
 
-// Define the query params for fetching header megamenu data.
+// Define the query params for fetching header megamenu data. Include referenced promo block data, if present
 export const megaMenuParams: QueryParams<null> = () => {
-  return queries.getParams().addInclude(['field_promo_reference'])
-}
-
-export const promoBlockParams: QueryParams<null> = () => {
   return queries
     .getParams()
-    .addInclude(['field_image', 'field_image.media', 'field_promo_link'])
+    .addInclude([
+      'field_promo_reference',
+      'field_promo_reference.field_image',
+      'field_promo_reference.field_image.image',
+      'field_promo_reference.field_promo_link',
+    ])
 }
 
 // Define the option types for the data loader.
 type DataOpts = QueryOpts<{
   params?: object
   megaMenuParams?: object
-  promoBlockParams?: object
 }>
```

Comment:
This should not be necessary. Our `data` function does not actually require any input `opts`. `params` and `megaMenuParams` are defined within this file and can be accessed directly as we now have it below. We should be able to remove this type definition (as well as its usage below; more commentary there).

Reply:
good callout, will nuke